packages/ckeditor5-html-support/src/conversionutils.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @license Copyright (c) 2003-2021, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module html-support/conversionutils
+ */
+
+import { cloneDeep } from 'lodash-es';
+
+/**
+* Helper function for downcast converter. Sets attributes on the given view element.
+*
+* @param {module:engine/view/downcastwriter~DowncastWriter} writer
+* @param {Object} viewAttributes
+* @param {module:engine/view/element~Element} viewElement
+*/
+export function setViewAttributes( writer, viewAttributes, viewElement ) {
+	if ( viewAttributes.attributes ) {
+		for ( const [ key, value ] of Object.entries( viewAttributes.attributes ) ) {
+			writer.setAttribute( key, value, viewElement );
+		}
+	}
+
+	if ( viewAttributes.styles ) {
+		writer.setStyle( viewAttributes.styles, viewElement );
+	}
+
+	if ( viewAttributes.classes ) {
+		writer.addClass( viewAttributes.classes, viewElement );
+	}
+}
+
+/**
+* Merges view element attribute objects.
+*
+* @param {Object} oldValue
+* @param {Object} newValue
+* @returns {Object}
+*/
+export function mergeViewElementAttributes( oldValue, newValue ) {
+	const result = cloneDeep( oldValue );
+
+	for ( const key in newValue ) {
+		// Merge classes.
+		if ( Array.isArray( newValue[ key ] ) ) {
+			result[ key ] = Array.from( new Set( [ ...oldValue[ key ], ...newValue[ key ] ] ) );
+		}
+
+		// Merge attributes or styles.
+		else {
+			result[ key ] = { ...oldValue[ key ], ...newValue[ key ] };
```

Comment:
The `newValue` / `oldValue` names don't seem to make sense here.

```suggestion
* @param {Object} attributesA
* @param {Object} attributesB
* @returns {Object}
*/
export function mergeViewElementAttributes( attributesA, attributesB ) {
	const result = cloneDeep( attributesA );

	for ( const key in attributesB ) {
		// Merge classes.
		if ( Array.isArray( attributesB[ key ] ) ) {
			result[ key ] = Array.from( new Set( [ ...attributesA[ key ], ...attributesB[ key ] ] ) );
		}

		// Merge attributes or styles.
		else {
			result[ key ] = { ...attributesA[ key ], ...attributesB[ key ] };
```

Reply:
Maybe we could go with `target` and `source` names instead? `attributesB` are replacing `attributesA`, so it seems like a common naming convention.